src/main/java/database/StorageDealerImpl.java

```diff
@@ -0,0 +1,21 @@
+package database;
+
+import java.util.Map;
+
+public class StorageDealerImpl implements StorageDealer {
+    @Override
+    public void checkBalance(Map<String, Integer> transactions) {
```

Comment:
It's a code quality issue. The name of the method `checkBalance` is not very clear about what it does. A better name for this method could be `validateBalance` or `ensureNonNegativeBalance`, because it checks if the balance is non-negative and throws an exception otherwise.